browser/static/js/cursor.js

```diff
@@ -0,0 +1,220 @@
+// NOTE: cursor and viewer are more abstract than the brush,
+// how they should relate to each other is an open question.
+// eg, should scale/dimensions be attributes only of viewer? etc
+
+// cursor to handle mouse movement and attributes
+class CalibanCursor {
+  constructor(width, height, scale) {
+    this.width = width
+    this.height = height
+    this.scale = scale;
+    this.pressed = false;
+    this.trace = [];
+
+    // mouse coords
+    // mouse position on canvas, no adjustment for padding
+    this.rawX = 0;
+    this.rawY = 0;
+    // adjusted for padding
+    this.canvasPosX = -5;
+    this.canvasPosY = -5;
+    // coordinates in original image (used for actions, labels, etc)
+    this.imgX;
+    this.imgY;
+    // what imgX and imgY were upon most recent click
+    this.storedClickX;
+    this.storedClickY;
+
+    this.label = 0;
+    // store as part of object to be able to get current label
+    this._segArray;
+  }
+
+  get segArray() {
+    return this._segArray;
+  }
+
+  set segArray(newSegArray) {
+    this._segArray = newSegArray;
+    if (this.inRange()) {
+      this.label = Math.abs(this.segArray[this.imgY][this.imgX]);
+    } else {
+      this.label = 0;
+    }
+
+  }
+
+  // check if the mouse position in canvas matches to a displayed part of image
+  inRange() {
+    if (this.canvasPosX >= 0 && this.canvasPosX < this.width * this.scale &&
+        this.canvasPosY >= 0 && this.canvasPosY < this.height * this.scale) {
+      return true;
+    } else {
+      return false;
```

Comment:
Just return the boolean condition instead of checking it.

```suggestion
    return (
      this.canvasPosX >= 0 && this.canvasPosX < this.width * this.scale &&
      this.canvasPosY >= 0 && this.canvasPosY < this.height * this.scale
    );
```